src/containers/SortingsTable.tsx

```diff
@@ -27,48 +28,36 @@ interface OwnProps {
 
 type Props = StateProps & DispatchProps & OwnProps
 
+const calculationPool = createCalculationPool({maxSimultaneous: 6})
+
 const SortingsTable: FunctionComponent<Props> = ({ sortings, onDeleteSortings, onSetSortingInfo, documentInfo }) => {
     const hither = useContext(HitherContext)
     const { documentId, feedUri, readOnly } = documentInfo;
-
-    function sortByKey<T extends {[key: string]: any}>(array: T[], key: string): T[] {
-        return array.sort(function (a, b) {
-            var x = a[key]; var y = b[key];
-            return ((x < y) ? -1 : ((x > y) ? 1 : 0));
-        });
-    }
-
-    const sortings2: Sorting[] = sortByKey<Sorting>(sortings, 'sortingLabel')
+    const infosInProgress = useRef(new Set<string>())
 
     useEffect(() => {
-        const thisSortings = sortings
-        ;(async () => {
-            for (const sor of sortings) {
-                if (thisSortings !== sortings) return
-                if (!sor.sortingInfo) {
-                    let info;
-                    try {
-                        // for a nice gui effect
-                        const sortingInfoJob = hither.createHitherJob(
-                            'createjob_get_sorting_info',
-                            { sorting_object: sor.sortingObject, recording_object: sor.recordingObject },
-                            {
-                                useClientCache: true
-                            }
-                        )
-                        info = await sortingInfoJob.wait() as SortingInfo;
-                        onSetSortingInfo({ sortingId: sor.sortingId, sortingInfo: info });
-                    }
-                    catch (err) {
-                        console.error(err);
-                        return;
+        for (const sor of sortings) {
```

Comment:
Cleaned up this useEffect to avoid excessive redundant hither job creation (was causing page hanging). The infosInProgress ref is necessary and seems to work, but there is probably a more elegant way of doing this.